src/Likvido.ApplicationInsights.Telemetry/HttpResponse/HttpResponseTelemetryInitializer.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace Likvido.ApplicationInsights.Telemetry
+{
+    /// <summary>
+    /// Enrich telemetry with response content.
+    /// </summary>
+    public class HttpResponseTelemetryInitializer : ITelemetryInitializer
+    {
+        private readonly Func<HttpResponseTelemetryConditionInfo, bool> _condition;
+
+        public HttpResponseTelemetryInitializer(Func<HttpResponseTelemetryConditionInfo, bool> condition)
+        {
+            _condition = condition ?? throw new ArgumentNullException(nameof(condition));
+        }
+
+        public void Initialize(ITelemetry telemetry)
+        {
+            try
+            {
+                if (telemetry is DependencyTelemetry dependencyTelemetry)
+                {
+                    Initialize(dependencyTelemetry);
+                }
+            }
+            catch
+            {
+                // Do not fail initializer.
+            }
+        }
+
+        private void Initialize(DependencyTelemetry telemetry)
+        {
+            // Check if telemetry is for http response.
+            if (telemetry.Type != "Http" ||
+                !telemetry.TryGetOperationDetail("HttpResponse", out var httpResponseObj) ||
+                !(httpResponseObj is HttpResponseMessage httpResponse))
+            {
+                return;
+            }
+
+            TryEnrichWithContentProperty(
+                telemetry,
+                httpResponse.StatusCode,
+                HttpResponseCustomProperty.ResponseContent,
+                httpResponse.Content);
+
+            TryEnrichWithContentProperty(
+                telemetry,
+                httpResponse.StatusCode,
+                HttpResponseCustomProperty.RequestContent,
+                httpResponse.RequestMessage.Content);
+        }
+
+        private void TryEnrichWithContentProperty(
+            DependencyTelemetry telemetry,
+            HttpStatusCode statusCode,
+            HttpResponseCustomProperty property,
+            HttpContent content)
+        {
+            if (_condition(new HttpResponseTelemetryConditionInfo(telemetry, property, statusCode)) &&
+                TryReadContentString(content, out var contentString))
+            {
+                telemetry.AddChunckedCustomProperty(property.ToString(), contentString!);
+            }
+        }
+
+        private static bool TryReadContentString(HttpContent content, out string? contentString)
+        {
+            if (content != null)
+            {
+                try
+                {
+                    contentString = content.ReadAsStringAsync().GetAwaiter().GetResult();
```

Comment:
this looks a bit dangerous to me. Not sure if the content stream can be read many times. So probably after this operation, they won't be usable.
It would be nice to have a link to some documentation about this

Reply:
No documentation was found, unfortunately. Option to get response was added here https://github.com/microsoft/ApplicationInsights-dotnet-server/issues/900, but I couldn't find more info.

`Initialize` is called sometime right before returning the response message to the caller, and it's the same object, so:

1. Content received in `Initialize`:
![image](https://user-images.githubusercontent.com/10668361/133143196-986fb2c0-6a37-40ce-b0ae-7c8d40ebec32.png)

2. Handling in service. The same response content is received here. It is now buffered as it was read in `Initialize` and is still readable:
![image](https://user-images.githubusercontent.com/10668361/133143287-c86da9be-e118-4da8-a329-7a1590d8569a.png)

So it seems the content can still be read several times.

Reply:
Maybe I'm too cautious but 
1. Check this guy. A really quick search gave him. He replaces and rewinds the streams. https://jarz.net/posts/app-insights-logging-middleware/
2. On your screenshot I can see `LimitMemoryStream`. I don't know how it works.

Anyway, if you are sure let's add this telemetry and try somewhere on the test server for a longer period than usual. Add ability to enable/disable it quickly from configuration